packages/store/src/decorators/state.ts

```diff
@@ -1,42 +1,42 @@
-import { ensureStoreMetadata } from '../internal/internals';
-import { StoreOptions, META_KEY } from '../symbols';
-
-const stateNameRegex = new RegExp('^[a-zA-Z0-9_]+$');
-
-/**
- * Error message
- * @ignore
- */
-export const stateNameErrorMessage = (name: string) =>
-  `${name} is not a valid state name. It needs to be a valid object property name.`;
+import { ensureStoreMetadata, MetaDataModel, StateClass } from '../internal/internals';
+import { META_KEY, META_OPTIONS_KEY, StoreOptions } from '../symbols';
+import { StoreValidators } from '../utils/store-validators';
+
+interface MutateMetaOptions<T> {
+  meta: MetaDataModel;
+  inheritedStateClass: StateClass;
+  optionsWithInheritance: StoreOptions<T>;
+}
 
 /**
  * Decorates a class with ngxs state information.
  */
 export function State<T>(options: StoreOptions<T>) {
-  return function(target: any) {
-    const meta = ensureStoreMetadata(target);
-
-    // Handle inheritance
-    if (Object.getPrototypeOf(target).hasOwnProperty(META_KEY)) {
-      const parentMeta = Object.getPrototypeOf(target)[META_KEY];
-
-      meta.actions = {
-        ...meta.actions,
-        ...parentMeta.actions
-      };
+  function getStateOptions(stateClass: StateClass): StoreOptions<T> {
```

Comment:
```suggestion
  function getStateOptions(inheritedStateClass: StateClass): StoreOptions<T> {
```
Rename this parameter too for consistency.